### PR TITLE
vision filtering and single tag gyro reprojection but from the correct branch

### PR DESCRIPTION
**Pull request by PotmanNob**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Vision/Vision.java [hunk 1]
```diff
@@ -1,18 +1,45 @@
 package frc.robot.Subsystems.Vision;
 
+import static edu.wpi.first.units.Units.DegreesPerSecond;
 import static frc.robot.GlobalConstants.ROBOT_MODE;
-import static frc.robot.Subsystems.Vision.VisionConstants.*;
-
+import static frc.robot.Subsystems.Vision.VisionConstants.APRIL_TAG_FIELD_LAYOUT;
```
Comment (GreenTomato5) [inline src/main/java/frc/robot/subsystems/Vision/Vision.java:5]: bro forgot about * 😭 you do NOT need 20 lins of static import just do
```java
import static frc.robot.Subsystems.Vision.VisionConstants.*;
```

File: src/main/java/frc/robot/subsystems/Vision/Vision.java [hunk 2]
```diff
@@ -36,9 +71,9 @@ public static Vision getInstance() {
 					};
 					case SIM -> new VisionIO[] {
 						new VisionIOPhotonVisionSim(FRONT_LEFT_CAM_NAME, ROBOT_TO_FRONT_LEFT_CAMERA, Drive.getInstance()::getPose),
-						new VisionIOPhotonVisionSim(FRONT_RIGHT_CAM_NAME, ROBOT_TO_FRONT_RIGHT_CAMERA, Drive.getInstance()::getPose),
-						new VisionIOPhotonVisionSim(BACK_LEFT_CAM_NAME, ROBOT_TO_BACK_LEFT_CAMERA, Drive.getInstance()::getPose),
-						new VisionIOPhotonVisionSim(BACK_RIGHT_CAM_NAME, ROBOT_TO_BACK_RIGHT_CAMERA, Drive.getInstance()::getPose),
+						// new VisionIOPhotonVisionSim(FRONT_RIGHT_CAM_NAME, ROBOT_TO_FRONT_RIGHT_CAMERA, Drive.getInstance()::getPose),
```
Comment (GreenTomato5) [inline src/main/java/frc/robot/subsystems/Vision/Vision.java:74]: un comment for real life, send video of single tag repro working vs with no repro and vision filtering working if possible.

Comment (GreenTomato5) [reply src/main/java/frc/robot/subsystems/Vision/Vision.java:74]: (for OA clipfarm/edge case test)

File: src/main/java/frc/robot/subsystems/Vision/Vision.java [hunk 3]
```diff
@@ -165,11 +196,51 @@ public void periodic() {
...
+			return VecBuilder.fill(xyStds, xyStds, Units.degreesToRadians(50)); // I dont even know, ts so random
+		} else {
+			xyStds = 0.5;
+			degStds = 6;
+			return VecBuilder.fill(xyStds, xyStds, degStds);
+		}
+	}
 
-		// Log summary data
-		Logger.recordOutput("Vision/Summary/TagPoses", allTagPoses.toArray(new Pose3d[allTagPoses.size()]));
```
Comment (GreenTomato5) [inline src/main/java/frc/robot/subsystems/Vision/Vision.java:170]: why is this logging gone? it was highkey helpful 😢

File: src/main/java/frc/robot/subsystems/Vision/VisionConstants.java [hunk 4]
```diff
@@ -52,17 +58,24 @@ public class VisionConstants {
...
+	public static final Set<Short> RED_REEF_TAGS = Set.of((short) 6, (short) 7, (short) 8, (short) 9, (short) 10, (short) 11);
+	public static final Set<Short> BLUE_REEF_TAGS = Set.of((short) 17, (short) 18, (short) 19, (short) 20, (short) 21, (short) 22);
+	public static final Set<Short> APRIL_TAG_IGNORE = Set.of((short) 14, (short) 15, (short) 4, (short) 5);
+
 	// AKIT TEMPLATE STUFF
 
 	// Basic filtering thresholds
 	public static final double maxAmbiguity = 0.3;
 	public static final double maxZError = 0.75;
+	public static final Angle GYRO_REPROJECTION_MARGIN = Degree.of(5);
```
Comment (GreenTomato5) [inline src/main/java/frc/robot/subsystems/Vision/VisionConstants.java:77]: is this what megatag2 uses? if u dnt kno ask bowan what his values r

**Review by GreenTomato5 — CHANGES_REQUESTED** [review]
You ate ts up young buck, changes are minor

**Review by GreenTomato5 — APPROVED** [review]
lgtm if it works in sim

File: src/main/java/frc/robot/subsystems/Vision/Vision.java [hunk 5]
```diff
@@ -1,26 +1,48 @@
...
+	private final SendableChooser<Boolean> reprojection = new SendableChooser<Boolean>();
+
+	List<Pose3d> allTagPoses = new LinkedList<>();
+	List<Pose3d> allRobotPoses = new LinkedList<>();
+	List<Pose3d> allRobotPosesAccepted = new LinkedList<>();
+	List<Pose3d> allRobotPosesRejected = new LinkedList<>();
+
+
+	Boolean isRedAlliance = DriverStation.getAlliance().isPresent() && DriverStation.getAlliance().get() == Alliance.Red;
+	Set<Short> allianceReefTag = isRedAlliance ? RED_REEF_TAGS : BLUE_REEF_TAGS;
```
Comment (GreenTomato5) [inline src/main/java/frc/robot/subsystems/Vision/Vision.java:45]: This is gona race condition, u have to update on televand auto init

**Review by GreenTomato5 — CHANGES_REQUESTED** [review]

**Review by GreenTomato5 — APPROVED** [review]
